### PR TITLE
fix: [CI-6082]: Add enforcement data via config till enforcement API’…

### PR DESCRIPTION
**Pull request by smjt-h**
…s are ready

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40235)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuildImpl.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          executionLimitSpec = ciExecutionServiceConfig.getExecutionLimits().getEnterprise();
+          break;
+        default:
+          executionLimitSpec = ciExecutionServiceConfig.getExecutionLimits().getFree();
+      }
+      long currExecutionCount = queueExecutionUtils.getActiveExecutionsCount(accountId);
+      long macExecutionsCount = queueExecutionUtils.getActiveExecutionsCount(accountId);
+      if (currExecutionCount < executionLimitSpec.getDefaultTotalExecutionCount()
+          && macExecutionsCount < executionLimitSpec.getDefaultMacExecutionCount()) {
+        return true;
```
Comment (aman-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuildImpl.java:42]: return currExecutionCount < executionLimitSpec.getDefaultTotalExecutionCount()
          && macExecutionsCount < executionLimitSpec.getDefaultMacExecutionCount();

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuild.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package io.harness.ci.enforcement;
+
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+@OwnedBy(HarnessTeam.CI)
+public interface CIExecutionBuild {
+  boolean validateBuildEnforcement(String accountID);
```
Comment (aman-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuild.java:8]: maybe `checkEnforcement`

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuild.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package io.harness.ci.enforcement;
+
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+@OwnedBy(HarnessTeam.CI)
+public interface CIExecutionBuild {
```
[the remark this answers is not in the document]
Comment (aman-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuild.java:7]: CIExecutionCountValidator maybe??

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuildImpl.java (hunk 1 above)
Comment (smjt-h) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/enforcement/CIExecutionBuildImpl.java]: done

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 7 files at r1, 3 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40235#-:-NHO_7ezEnXAv57YOxB6:bnfp4nl)* status: 1 file(s) to review.

**Comment by smjt-h** [comment]
trigger ti2

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40235#-:-NHc_korEDPzLyWDHU_O:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**